src/components/saveChangesModal/index.js

```diff
@@ -0,0 +1,91 @@
+import { useNavigate } from "react-router-dom"
+import { useState } from "react";
+import useModal from "../../hooks/useModal"
+import './style.css'
+import Button from '../button'
+import Toast from '../toast'
+import { patch } from "../../service/apiClient";
+import ErrorMessage from "../errorMessage";
+
+const SaveChangesModal = ({ formState, id, loggedInUserInfo }) => {
+    const [isError, setIsError] = useState(false)
+    const { closeModal } = useModal()
+    const navigate = useNavigate()
+    const handleDontSave = () => {       
+            closeModal()
+            navigate(`/profile/${id}`)
+    }
+
+    const handleCancel = () => {
+        closeModal()
+        navigate(`/profile/${id}/edit`)
+    }
+
+    const handleSave = () => {
+        handleSubmit()
+        if(isError === true){
+            
+        }
+        else{
+            closeModal()
+            navigate(`/profile/${id}`)
+        }
+        return (
+            <Toast text="profile saved!" linkText="Edit" linkTo="/profile/edit" />
+        )
+    }
+    const handleSubmit = () => {
+        const editedProfile = formState
+        const formDataPATCH = async () => {
+            const endpoint = `users/${id}`
+            const data = {
+                email: editedProfile.email,
+                password: editedProfile.password,
+                firstName: editedProfile.firstName,
+                lastName: editedProfile.lastName,
+                biography: editedProfile.biography,
+                githubUrl: editedProfile.githubUrl
+            }
+            if (loggedInUserInfo.role === "TEACHER") {
+                data.role = editedProfile.role
+                data.cohortId = editedProfile.cohortId
+
+                patch(endpoint, editedProfile)
+                    .catch((error) => { setIsError(true) })
+
+            }
+            formDataPATCH()
```

Comment:
This `formDataPATCH` call is inside the `formDataPATCH` function, you'll end up with an endless recursive function calling itself until the app crashes if this runs